.template/variants/api/app/controllers/concerns/localization.rb

```diff
@@ -2,12 +2,26 @@ module Localization
   extend ActiveSupport::Concern
 
   included do
-    before_action :set_locale
+    around_action :switch_locale
 
     protected
 
-    def set_locale
-      I18n.locale = request.headers[:'Accept-Language'] || I18n.default_locale
+    def switch_locale(&action)
+      locale = extract_locale_from_header || I18n.default_locale
+
+      I18n.with_locale(locale, &action)
+    end
+
+    def extract_locale_from_header
+      return request.headers[:'Accept-Language'] if I18n.locale_available?(request.headers[:'Accept-Language'])
+      
+      nil
+    end
+
+    private
+
+    def default_url_options
```

Comment:
@andyduong1920 If we go for api version, we will not need `default_url_options` I guess

Reply:
yeah, good point, removed that in https://github.com/nimblehq/rails-templates/pull/241/commits/39ea1f87d09352c5bc8e5f1bda9b50ce16a6f3c0 @ankitkalia1195